src/smlp_py/smlp_plots.py

```diff
@@ -2,7 +2,12 @@
 # This file is part of smlp.
 
 import matplotlib
-matplotlib.use('TkAgg')  # Set the backend to 'TkAgg' before importing pyplot, to enable interactive plots
+import os
+if not os.environ.get("DISPLAY"):
+    matplotlib.use("Agg")  # Non-interactive backend
+else:
+    matplotlib.use("TkAgg") # Set the backend to 'TkAgg' before importing pyplot, to enable interactive plots
```

Comment:
I wish we wouldn't default to this kind of interactive use of SMLP based on some unrelated environment settings. I understand it can be useful, but should, in my opinion, at least be possible to deactivate without messing with envvars.